From 6bd717744c7b479b3f0f117fe38dfdba167f5082 Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Thu, 15 Sep 2011 22:40:10 +0800
Subject: [PATCH 158/280] ENGR00156800 vpu: Fix decoding mp4PackedPBFrame
 stream failure issue

Decoding the stream with mp4PackedPBFrame is failure due to write
point in bitstream set wrong.
This patch will fix it.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 vpu/vpu_lib.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/vpu/vpu_lib.c b/vpu/vpu_lib.c
index a5522fa..32d44b2 100644
--- a/vpu/vpu_lib.c
+++ b/vpu/vpu_lib.c
@@ -3552,6 +3552,9 @@ RetCode vpu_DecGetOutputInfo(DecHandle handle, DecOutputInfo * info)
 		info->notSufficientSliceBuffer = (val >> 2) & 0x1;
 	} else if (pCodecInst->codecMode == MP4_DEC) {
 		info->mp4PackedPBframe = ((val >> 16) & 0x01);
+		/* Need to backup WR_PTR for mp4PackedPBframe */
+		if (info->mp4PackedPBframe)
+			pCodecInst->ctxRegs[CTX_BIT_WR_PTR] = VpuReadReg(BIT_WR_PTR);
 	}
 
 	val = VpuReadReg(RET_DEC_PIC_SIZE);     /* decoding picture size */
-- 
1.8.0

